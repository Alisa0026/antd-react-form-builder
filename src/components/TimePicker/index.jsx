import React from 'react'
import _ from 'lodash'
import moment from 'moment'
import BasicItem from '../BasicItem'
import TimePicker from 'antd/lib/time-picker'
import FormItemComponentDecorator from '../../decorator/FormItemComponent'

function component(BasicItemComponent){
  @FormItemComponentDecorator
  class FTimePicker extends React.Component {

    render(){
      let { ...other } = this.props;

      other.targetComponent = TimePicker;
      if(_.isString(other.value)){
        other.value = new moment(other.value,"HH:mm:ss")
      }
      other.type = "timepicker";
      this.propsAdapter(other);
      return (
        <BasicItemComponent { ...other }/>
      )
    }
    
  }
  return FTimePicker;
}

export default component(BasicItem) 




