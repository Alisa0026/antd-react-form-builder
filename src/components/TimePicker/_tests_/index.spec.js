import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import moment from 'moment'
import LocaleProvider from 'antd/lib/locale-provider'
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import TimePicker from '../index'

describe('TimePicker',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <TimePicker 
        required
        className="test"
        label="name"
        name="test"
        value="10:10:00"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(TimePicker,"timepicker"); 
  it("stirng value should be rendered correctly",function(){
    const wrapper = mount(
      <TimePicker 
        required
        className="test"
        label="name"
        name="test"
        value="15:08:34"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(antdInputMount.props().value).toEqual(new moment("15:08:34","HH:mm:ss"));
  })
  it("moment value should be rendered correctly",function(){
    const wrapper = mount(
      <TimePicker 
        required
        className="test"
        label="name"
        name="test"
        value={ new moment("15:08:34","HH:mm:ss") }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(antdInputMount.props().value).toEqual(new moment("15:08:34","HH:mm:ss"));
  })
  it("should have hidden <Input /> ",function(){
    var value = new moment("15:08:34","HH:mm:ss");
    const wrapper = mount(
      <TimePicker 
        required
        className="test"
        label="name"
        name="test"
        value={ value }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(wrapper.childAt(1).props().value).toEqual(Math.floor(+value / 1000));
    expect(wrapper.childAt(1).props().name).toEqual('test');
  })
})

