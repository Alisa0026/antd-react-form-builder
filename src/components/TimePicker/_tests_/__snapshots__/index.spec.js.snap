// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <span
          class="ant-time-picker test ant-time-picker-default"
        >
          <input
            class="ant-time-picker-input"
            placeholder="请选择时间"
            readonly=""
            type="text"
            value=""
          />
          <span
            class="ant-time-picker-icon"
          />
        </span>
      </div>
    </div>
  </div>
  <input
    class="ant-input"
    name="test"
    type="hidden"
    value=""
  />
</span>
`;
